cmd/manager/main.go

```diff
@@ -97,9 +104,11 @@ func main() {
 		caCertDir                 string
 		globalPullSecret          string
 	)
-	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&metricsAddr, "metrics-bind-address", "0", "The address for the metrics endpoint. Use :8443 for HTTPS or set to 0 to disable the metrics server. Requires both tls-cert and tls-key.")
```

Comment:
Okay, so I see that in the accepted RFC you wrote that you specified that this would be set to 0 by default, which means that metrics are disabled by default.

Since that's the accepted change in the RFC, I would suggest at least tweaking the help text here so it doesn't sound like the flag only accepts either 8443 or 0--I know that we would *prefer* users use 8443 and that our manifest will use 8443 by default, but it's not technically true that this flag only accepts 8443 and 0, which is how it reads right now. 

Since it's set to 0 by default, you don't need to mention setting it to 0 manually. Simply omitting the flag disables the server, which means setting the flag at all enables it. So, the text should say something like "Enable the metrics server at the specified endpoint address. Requires both tls-cert and tls-key. Suggested value of \":8443\""

I do have reservations about this design of defaulting to 0 though. Since we do have a preferred port of :8443, it makes more sense to me to make the default be 8443, with help text like "The address for the metrics endpoint. Defaults to :8443. Requires both tls-cert and tls-key. Set to 0 to completely disable the metrics server." but even this seems like it's cramming too much functionality into a single flag since you're doing what the flag is named, setting the metrics bind address, as well as controlling if the server is enabled or disabled, AND it's dependent on the presence of 2 other flags. Complex flags like this can quickly become confusing to a user. Ideally you'd control if it's disabled/enabled with one flag, what the address is with another flag (the existing one), and check for the proper application of those flags together like we do with tls-key and tls-cert.

__
A separate concern I have is by changing the default from 8080 to 0, any previous usage of operator-controller that didn't specify a value for --metrics-bind-address but rather relied on the default value of 8080 will go from having a functioning server endpoint to having it disabled after this merge (not that there's tons of usage out there right now, but just in the spirit of not introducing breaking changes). That's a silent breaking change which won't error out to alert the user that core functionality is altered. Whereas, if you default to 8443, you keep the program running the same way the user intended, with the server enabled, but they'll get the proper error output alerting them to the fact that they now need to provide tls-cert and tls-key.

Reply:
>but it's not technically true that this flag only accepts 8443 and 0

Why the flag only accepts :8443 
Why could we not pass other value
We want / recommend :8443 but still possible use another right

Reply:
I updated the description.
Also, see that I fixed the map of the port
It should be done in the service
Let me know if that address your main concerns and if we can move forward.

Reply:
I agree with most of what @trgeiger mentioned. if the user can omit the flag then the default of `metrics-bind-address` flag should be 8443. Also have same concern about breaking change.

Reply:
Takeaways

- If cert/key are not provided, metrics-bind-addr is not allowed
- If cert/key are provided, and metrics-bind-addr is not, we default to :8443
- If cert/key are provided and metrics-bind-addr is also provided, we honor what is provided.

Reply:
So closing this one as sorted out